datafusion/physical-plan/src/joins/nested_loop_join.rs

```diff
@@ -828,13 +833,127 @@ impl<T: BatchTransformer> NestedLoopJoinStream<T> {
                     handle_state!(self.process_probe_batch())
                 }
                 NestedLoopJoinStreamState::ExhaustedProbeSide => {
-                    handle_state!(self.process_unmatched_build_batch())
+                    handle_state!(self.prepare_unmatched_output_indices())
+                }
+                NestedLoopJoinStreamState::OutputUnmatchedBuildRows(_) => {
+                    handle_state!(self.build_unmatched_output())
                 }
                 NestedLoopJoinStreamState::Completed => Poll::Ready(None),
             };
         }
     }
 
+    fn get_next_join_result(&mut self) -> Result<Option<RecordBatch>> {
+        let (left_indices, right_indices, start) =
+            self.join_result_status.as_mut().ok_or_else(|| {
+                datafusion_common::_internal_datafusion_err!(
+                    "should have join_result_status"
```

Comment:
I think we can change this to be more verbose, and we can use 
`internal_err!`
```suggestion
                internal_err!(
                    "get_next_join_result called without initializing join_result_status"
```

Reply:
> we can use internal_err!


`internal_err!` return `Result` but `ok_or_else` require closure return `error`

https://github.com/rust-lang/rust/blob/86e05cd300fac9e83e812c4d46582b48db780d8f/library/core/src/option.rs#L1334-L1336